### ISSUE TITLE
Flatten function

### ISSUE THREAD

**Issue by leandromoh** [morelinq/MoreLINQ#356]
I think would be useful a function that flatten nestead sequences. Not only `IEnumerable<IEnumerable<T>>`, or `IEnumerable<IEnumerable<IEnumerable<T>>>` but in any level of depth. 

So I propose the following function:

```
static IEnumerable<object> Flatten(this IEnumerable source)
{
	foreach (var e in source)
	{
		if (e is IEnumerable seq)
		{
			foreach (var i in Flatten(seq))
			{
				yield return i;
			}
		}
		else
		{
			yield return e;
		}
	}
}
```

So, we can flatten sequences in any deepth just with `Flatten(source).Cast<T>()`.
It also works with arbitrarily-nested sequence of different types. In the example below, 

`list.Flatten().OfType<int>() `returns `[1, 2, 3, 4, 5, 6]` and
`list.Flatten().OfType<string>()` returns `["foo", "bar"]`
`list.Flatten().OfType<bool>()` returns `[true]`
```
var list = new List<object>()
{
    1,
    2,
    new List<object>()
    {
        3,
        new ArrayList()
        {
            4,
            "foo"
        },
        5,
        true,
    },
   "bar",
    6
};
```

I would like to submit a PR for this.
What do you think?

**Comment by atifaziz** [comment]
> I would like to submit a PR for this.

Go ahead & thanks!

**Comment by atifaziz** [comment]
This is now available in release [2.10.0](https://github.com/morelinq/MoreLINQ/releases/tag/v2.10.0).
